twpca/utils.py

```diff
@@ -69,14 +70,11 @@ def compute_lowrank_factors(data, n_components, fit_trial_factors, last_idx, sca
         last_idx: nd-array, list of ints holding last index before trial end
         scale: scale neuron and time factors by this amount, default 1.0
     """
-    # do svd on trial-averaged data matrix
-    # TODO: use randomized/truncated SVD to speed this up
-    u, s, v = np.linalg.svd(np.nanmean(data, axis=0), full_matrices=False)
-    sqs = np.sqrt(s) * scale
-
-    # set neuron and time factors to top singular vectors
-    time_fctr = (u[:, :n_components] * sqs[:n_components]).astype(np.float32)
-    neuron_fctr = (v[:n_components].T * sqs[:n_components]).astype(np.float32)
+    # do matrix decomposition on trial-averaged data matrix
+    DecompModel = NMF if nonneg else TruncatedSVD
+    model = DecompModel(n_components=n_components)
+    time_fctr = model.fit_transform(np.nanmean(data, axis=0)).astype(np.float32)
+    neuron_fctr = np.transpose(model.components_).astype(np.float32)
```

Comment:
Probably not important, but is addressed in the next commit to be consistent with previous code.